modules/quirks/qualcomm/sdm845-modem.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.mobile.quirks.qualcomm;
+  inherit (lib) mkIf mkOption types;
+in
+{
+  options.mobile = {
+    quirks.qualcomm.sdm845-modem.enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable this on a mainline-based SDM845 device for modem support
+      '';
+    };
+  };
+  config = mkIf (cfg.sdm845-modem.enable) {
+    systemd.services = {
+      rmtfs = {
+        wantedBy = [ "multi-user.target" ];
+        requires = [ "qrtr-ns.service" ];
+        after = [ "qrtr-ns.service" ];
+        serviceConfig = {
+          ExecStart = "${pkgs.rmtfs}/bin/rmtfs -r -P -s";
+          Restart = "always";
+          RestartSec = "1";
+        };
+      };
+      qrtr-ns = {
+        serviceConfig = {
+          ExecStart = "${pkgs.qrtr}/bin/qrtr-ns -f 1";
+          Restart = "always";
+        };
+      };
+      tqftpserv = {
+        wantedBy = [ "multi-user.target" ];
+        requires = [ "qrtr-ns.service" ];
+        after = [ "qrtr-ns.service" ];
+        serviceConfig = {
+          ExecStart = "${pkgs.tqftpserv}/bin/tqftpserv";
+          Restart = "always";
+        };
+      };
+      pd-mapper = {
+        wantedBy = [ "multi-user.target" ];
+        requires = [ "qrtr-ns.service" ];
+        after = [ "qrtr-ns.service" ];
+        serviceConfig = {
+          ExecStart = "${pkgs.pd-mapper}/bin/pd-mapper";
+          Restart = "always";
+        };
+      };
+      msm-modem-uim-selection = {
```

Comment:
I was never able to get this to work. It got the modem to appear in the system with this and it read data about my SIM card. I never tested it on PostmarketOS either, so it may have been my carrier or something.

If I remember correctly, the other services were needed for WiFi, which I did get working.

Reply:
To be 100% clear: I got data over cellular working without issues within Phosh. I think everything is (now) working as much as supported by mainline for radios.